bin/build.js

```diff
@@ -0,0 +1,64 @@
+const path = require('path')
+const browserify = require('browserify')
+const resolve = require('resolve')
+const xtend = require('xtend')
+const fs = require('fs')
+const mkdirp = require('mkdirp')
+
+const defaults = {
+  optimize: false,
+  entry: '.',
+  html: {},
+  css: {},
+  js: {}
+}
+
+const cwd = process.cwd()
+
+// resolve a path according to require.resolve algorithm
+// string -> string
+function resolveEntryFile (relativePath) {
+  const first = relativePath.charAt(0)
+  const entry = ['.', '/'].includes(first) ? relativePath : './' + relativePath
+  return resolve.sync(entry, {basedir: cwd})
+}
+
+function build (options, cb) {
+  const bankai = require('../')({
+    optimize: true
+  })
+
+  const settings = xtend({}, defaults, options)
+  const callback = cb || function () {}
+
+  const entryFile = resolveEntryFile(settings.entry)
+  const outputDir = settings.dir
+
+  // Register css & html if specified. Register js no matter what
+  const css = settings.css && bankai.css(settings.css)
+  const html = settings.html && bankai.html(settings.html)
+  const js = bankai.js(browserify, entryFile, settings.js)
+
+  mkdirp(outputDir, (err) => {
+    if (err) return console.error(`Error creating directory ${outputDir}`, err)
+
+    if (js) {
+      const jsPath = path.join(outputDir, settings.html.js || 'bundle.js')
+      js().pipe(fs.createWriteStream(jsPath))
+    }
+
+    if (css) {
+      const cssPath = path.join(outputDir, settings.html.css || 'bundle.css')
+      css().pipe(fs.createWriteStream(cssPath))
+    }
+
+    if (html) {
+      const htmlPath = path.join(outputDir, 'index.html')
+      html().pipe(fs.createWriteStream(htmlPath))
+    }
+  })
+
+  callback()
```

Comment:
hmm, so this callback doesn't really do anything in this case; as it'll most likely be called before the bundles are done resolving. 

Something we could do is wrap the streams in a function, push the functions to an array and resolve them in parallel, calling the callback when done. Something like this:
```js
const parallel = require('run-parallel')
const pump = require('pump')

function build (opts, cb) {

  /* whole
   * lot
   * of
   * stuff
   */

  mkdirp(outputDir, (err) => {
    const fns = []

    if (js) {
      fns.push(function (done) {
        const jsPath = path.join(outputDir, settings.html.js || 'bundle.js')
        pump(js, fs.createWriteStream(jsPath), done)
      })
    }

    if (css) {
      fns.push(function (done) {
        const cssPath = path.join(outputDir, settings.html.css || 'bundle.css')
        pump(css, fs.createWriteStream(cssPath), done)
      })
    }

    if (html) {
      fns.push(function (done) {
        const cssPath = path.join(outputDir, 'index.html')
        pump(html, fs.createWriteStream(htmlPath), done)
      })
    }

    parallel(fns, cb)
  })
}
```
You reckon?